docs/architecture/speculative-workflow-task.md

```diff
@@ -6,157 +6,168 @@ There are three types of Workflow Task:
   2. Transient
   3. Speculative
 
-Every Workflow Task ships history to the worker. Last events must be `WorkflowTaskScheduled`
-and `WorkflowTaskStarted`. There might be some events in between if they come after Workflow Task
-was scheduled but not started (e.g., Workflow worker was down and didn't poll for Workflow Task).
-
-**Normal Workflow Task** is created by server every time when server needs Workflow to make progress.
-If Workflow Task fails (worker responds with call to `RespondWorkflowTaskFailed` or an error occurred while
-processing `RespondWorkflowTaskCompleted`) or times out (worker is disconnected),
-server writes corresponding Workflow Task failed event in the history and increase attempt count
-in the mutable state. For the next attempt, Workflow Task events (`WorkflowTaskScheduled`
-and `WorkflowTaskStarted`) are not written into the history but attached to the response of
-`RecordWorkflowTaskStarted` API. These are transient Workflow Task events. Worker is not aware of any "transience"
-of these events. If Workflow Task keeps failing, attempt counter is getting increased in mutable state,
-but no new fail events are written into the history and new transient Workflow Task events are just recreated.
-Workflow Task, which has transient Workflow Task events, is called **transient Workflow Task**.
-When Workflow Task finally completes, `WorkflowTaskScheduled` and `WorkflowTaskStarted` events
-are getting written to history followed by `WorklfowTaskCompleted` event.
+Every Workflow Task ships history events to the worker. It must always contain the two events
+`WorkflowTaskScheduled` and `WorkflowTaskStarted`. There might be some events in-between them if
+they came in after the Workflow Task was scheduled but not yet started (e.g. when the Workflow worker
+was down and didn't poll for Workflow Task).
+
+A **normal Workflow Task** is created by the server when it needs a Workflow to make progress. If 
+the Workflow Task fails (i.e. worker responds with a call to `RespondWorkflowTaskFailed` or an error
+occurred while processing `RespondWorkflowTaskCompleted`) or times out (e.g. worker is disconnected),
+the server writes a corresponding Workflow Task failed event to the history and increases the
+attempt count in the mutable state.
+
+For the next attempt, a **transient Workflow Task** is used: the Workflow Task events
```

Comment:
So... the is idea here was that there is no such thing as *transient* WFT. All WFT are the same. There are transient *events*. And we call WFT with such events "transient WFT" just for convenience instead of saying "WFT with transient events". I think this idea is lost in your version, although it still describes it pretty well.

Reply:
👍 I'll clarify